src/main/java/com/exactpro/th2/fix/client/Main.java

```diff
@@ -333,11 +333,14 @@ protected Event initialize() {
                         }
 
                         if (sessionSettings.getOrderingFields() != null && sessionSettings.getOrderingFields().equals(YES_SETTING)) {
-                            fixMessage = messageFactory.create(MessageUtils.getMessageType(strMessage), dataDictionary.getOrderedFields());
+                            String msgType = MessageUtils.getMessageType(strMessage);
+                            fixMessage = messageFactory.create(msgType, dataDictionary.getMsgTypeOrderedFields(msgType),
+                                    dataDictionary.getMsgTypeOrderedFields("HEADER"), dataDictionary.getMsgTypeOrderedFields("TRAILER"));
                             fixMessage.fromString(strMessage, sessionDataDictionary, dataDictionary, true);
                         } else {
                             fixMessage = new FixMessage(strMessage, sessionDataDictionary, dataDictionary);
                         }
+                        dataDictionary.validate(fixMessage, true);
```

Comment:
> true

Why?

Reply:
to validate a header in FIX messages > 5.0 version, we need 2 dictionaries, but the validate method in qfj uses only one, so errors will occur when validating the header